R/derive_param_response.R

```diff
@@ -0,0 +1,248 @@
+#'
+#'   A list of symbols created using `vars()` is expected.
+#'
+#' @details
+#'
+#'   1. The Date of the end of the assessment period (e.g. Progressive disease,
+#'    as defined by `pd_source`) is added to the response dataset.
+#'
+#'   1. The response dataset is restricted to observations occurring before
+#'   ** or on ** the date of progressive disease.
+#'
+#'   1. For each subject (with respect to the variables specified for the
+#'   `subject_keys` parameter), the first observation (with respect to
+#'   `ADT`) where the response condition (`filter_source` parameter) is
+#'   fulfilled is selected.
+#'
+#'   1. For each observation in `dataset_adsl` a new observation is created.
+#'   + For subjects with a response `AVALC` is set to `"Y"`, `AVAL` to `1`,
+#'   and `ADT` to  the first date (`ADT`) where the response condition is
+#'   fulfilled.
+#'   + For all other subjects `AVALC` is set to `"N"`, `AVAL` to `0`
+#'   and `ADT` to `NA`.
+#'
+#'   1. The variables specified by the `set_values_to` parameter are added to
+#'   the new observations.
+#'
+#'   1. The new observations are added to input dataset.
+#'
+#' @author Samia Kabi
+#'
+#' @return The input dataset with a new parameter indicating if and when a
+#' response occurred
+#'
+#' @keywords derivation bds
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(admiral)
+#' library(lubridate)
+#' library(rlang)
+#'
+#' adsl <- tibble::tribble(
+#'   ~USUBJID,
+#'   "1",
+#'   "2",
+#'   "3",
+#'   "4"
+#' ) %>%
+#'   mutate(STUDYID = "XX1234")
+#'
+#' adrs0 <- tibble::tribble(
+#'   ~USUBJID, ~ADTC, ~AVALC,
+#'   "1", "2020-01-02", "PR",
+#'   "1", "2020-02-01", "CR",
+#'   "1", "2020-03-01", "CR",
+#'   "1", "2020-04-01", "SD",
+#'   "2", "2021-06-15", "SD",
+#'   "2", "2021-07-16", "PD",
+#'   "2", "2021-09-14", "PD",
+#'   "3", "2021-09-14", "SD",
+#'   "3", "2021-10-30", "PD",
+#'   "3", "2021-12-25", "CR"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "XX1234",
+#'     ADT = ymd(ADTC),
+#'     PARAMCD = "OVR",
+#'     PARAM = "Overall Response",
+#'     ANL01FL = "Y"
+#'   ) %>%
+#'   select(-ADTC)
+#'
+#' # Add a parameter for the date of Progressive disease (PARAMCD = PD)
+#' adrs1 <- derive_param_first_event(
+#'   dataset = adrs0,
+#'   dataset_adsl = adsl,
+#'   dataset_source = adrs0,
+#'   filter_source = PARAMCD == "OVR" & AVALC == "PD",
+#'   date_var = ADT,
+#'   set_values_to = vars(
+#'     PARAMCD = "PD",
+#'     PARAM = "Disease Progression",
+#'     ANL01FL = "Y"
+#'   )
+#' )
+#'
+#' # Define the end of the assessment period for responses:
+#' # all responses before or on the first PD will be used.
+#' pd <- date_source(
+#'   dataset_name = "adrs1",
+#'   date = ADT,
+#'   filter = PARAMCD == "PD" & AVALC == "Y"
+#' )
+#' # Derive the response parameter
+#' derive_param_response(
+#'   dataset = adrs1,
+#'   dataset_adsl = adsl,
+#'   filter_source = PARAMCD == "OVR" & AVALC %in% c("CR", "PR"),
```

Comment:
please include `& ANL01FL == "Y"` - see my explanation for this in filter_pd PR review comment